src/WindowsPackageManager/main.cpp

```diff
@@ -14,13 +14,14 @@
 
 using namespace winrt::Microsoft::Management::Deployment;
 
-// CreatorMap for out-of-proc com registration
+// CreatorMap for out-of-proc com registration and direct in-proc com class construction
 CoCreatableClassWrlCreatorMapInclude(PackageManager);
 CoCreatableClassWrlCreatorMapInclude(FindPackagesOptions);
 CoCreatableClassWrlCreatorMapInclude(CreateCompositePackageCatalogOptions);
 CoCreatableClassWrlCreatorMapInclude(InstallOptions);
 CoCreatableClassWrlCreatorMapInclude(UninstallOptions);
 CoCreatableClassWrlCreatorMapInclude(PackageMatchFilter);
+CoCreatableClassWrlCreatorMapInclude(PackageManagerSettings);

```

Comment:
I feel like the only thing stopping this from being used OOP is the fact that it isn't registered in our `AppxManifest.xml`.  I don't know if that makes it impossible to create, or just harder.

Not that it would be the end of the world or anything, but I think we would prefer that they not create it OOP if that doesn't require a lot of work.

Reply:
After several hours of exploring, I think it requires some amount of work to do custom registration and maintaining the registrations. And it needs more thorough testing if we choose that route. So I'll leave it for now.